### ISSUE TITLE
Error: protected method 'object_id' called for Mocks::Registry::LastArgsKey

### ISSUE THREAD

**Issue by rurounijones** [waterlink/mocks.cr#45]
Very new to Crystal so apologies if needed information not provided. Let me know if anything else is needed.

```
crystal -v
Crystal 0.32.1 [41bd18fbe] (2019-12-18)
LLVM: 8.0.0
Default target: x86_64-unknown-linux-gnu
```
Adding mocks.cr to dependencies and adding `require "mocks/spec"` to the spec_helper.cr results in the following:

```
> $ crystal spec                                                                                                                                                                                     

Showing last frame. Use --error-trace for full trace.

In /usr/share/crystal/src/hash.cr:907:18

 907 | hash = key.object_id.hash.to_u32!
                  ^--------
Error: protected method 'object_id' called for Mocks::Registry::LastArgsKey
```

Full error trace:

```
crystal spec --error-trace                                                                                                                                                                       
error in line 1
Error: while requiring "./spec/tacview_client_spec.cr"


In spec/tacview_client_spec.cr:1:1

 1 | require "./spec_helper"
     ^
Error: while requiring "./spec_helper"


In spec/spec_helper.cr:2:1

 2 | require "mocks/spec"
     ^
Error: while requiring "mocks/spec"


In lib/mocks/src/spec.cr:10:9

 10 | Mocks.reset
            ^----
Error: instantiating 'Mocks:Module#reset()'


In lib/mocks/src/mocks.cr:15:5

 15 | reset_registries
      ^---------------
Error: instantiating 'reset_registries()'


In lib/mocks/src/mocks/registry.cr:9:38

 9 | Registry::Method::LAST_ARGS.keys.each do |key|
                                      ^---
Error: instantiating 'Array(Mocks::Registry::LastArgsKey)#each()'


In /usr/share/crystal/src/indexable.cr:187:5

 187 | each_index do |i|
       ^---------
Error: instantiating 'each_index()'


In /usr/share/crystal/src/indexable.cr:187:5

 187 | each_index do |i|
       ^---------
Error: instantiating 'each_index()'


In lib/mocks/src/mocks/registry.cr:9:38

 9 | Registry::Method::LAST_ARGS.keys.each do |key|
                                      ^---
Error: instantiating 'Array(Mocks::Registry::LastArgsKey)#each()'


In lib/mocks/src/mocks/registry.cr:10:35

 10 | Registry::Method::LAST_ARGS.delete(key)
                                  ^-----
Error: instantiating 'Hash(Mocks::Registry::LastArgsKey, String)#delete(Mocks::Registry::LastArgsKey)'


In /usr/share/crystal/src/hash.cr:1195:5

 1195 | delete(key) { nil }
        ^-----
Error: instantiating 'delete(Mocks::Registry::LastArgsKey)'


In /usr/share/crystal/src/hash.cr:1207:13

 1207 | entry = delete_impl(key)
                ^----------
Error: instantiating 'delete_impl(Mocks::Registry::LastArgsKey)'


In /usr/share/crystal/src/hash.cr:425:12

 425 | hash = key_hash(key)
              ^-------
Error: instantiating 'key_hash(Mocks::Registry::LastArgsKey)'


In /usr/share/crystal/src/hash.cr:907:18

 907 | hash = key.object_id.hash.to_u32!
                  ^--------
Error: protected method 'object_id' called for Mocks::Registry::LastArgsKey
```

**Comment by taylorthurlow** [comment]
This is because the shard is not compatible with the latest versions of Crystal. A user [has a fork](https://github.com/DocSpring/mocks.cr/tree/updates_for_0.31.1) updated for `0.31.1`, but even then, you'll get this error if you use `0.32.1`.

**Comment by iambudi** [comment]
Any knows fix for 0.35.1 or alternative shards?
